src/ios/GeolocationPlugin.m

```diff
@@ -0,0 +1,276 @@
+//
+//  GeolocationPlugin.m
+//  Geolocation
+//
+//  Created by dev on 2/9/16.
+//
+//
+
+#import "GeolocationPlugin.h"
+
+@implementation GeolocationPlugin
+
+-(void)start:(CDVInvokedUrlCommand *)command
+{
+    coord1 = [command.arguments objectAtIndex:0];   // Get center and radio of region1
+    coord2 = [command.arguments objectAtIndex:1];   // Get center and radio of region2
+    url = [command.arguments objectAtIndex:2];      // Get server url
+
+    
+    [self stopTimer];
+    [self startTimedTask];
+}
+
+-(void)startTimedTask{
+    time_interval = [NSTimer scheduledTimerWithTimeInterval:75.0 target:self selector:@selector(performBackgroundTask) userInfo:nil repeats:YES];
+}
+-(void)performBackgroundTask{
+    
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+        dispatch_async(dispatch_get_main_queue(), ^{
+            [self initData];
+        });
+    });
+}
+
+-(void)stopTimer{
+    if(time_interval)
+    {
+        [time_interval invalidate];
+        time_interval = nil;
+        [self stopUpdatingLocationWithMessage:NSLocalizedString(@"Error", @"Error")];
+    }
+}
+
+-(void)stop:(CDVInvokedUrlCommand *)command
+{
+    [self stopTimer];
+}
+
+-(void)initData{
+    [self createDataBase];
+    
+    // Add location Manager class
+    if (nil == self.locationManager) {
+        self.locationManager = [[CLLocationManager alloc] init];
+        // Get version of iOS
+        if ([[[UIDevice currentDevice] systemVersion] floatValue] >= 8.0) {
+            
+            [ self.locationManager requestWhenInUseAuthorization];
+            
+            [self.locationManager requestAlwaysAuthorization];
+        }
+    }
+    now     = [NSDate date];
+    now1    = [NSDate date];
+    
+    if (nil == self.locationManager)
+        self.locationManager = [[CLLocationManager alloc] init];
+    self.locationManager.delegate = self;
+    
+    
+    [self.locationManager startMonitoringSignificantLocationChanges];
+    [self.locationManager startUpdatingLocation];
+    
+    // add multi region
+    [self regionStart:coord1];
+    [self regionStart:coord2];
+}
+
+
+// recieve GPS updated location
+- (void)locationManager:(CLLocationManager *)manager didUpdateLocations:(NSArray *)locations {
+    
+    location = [locations lastObject];
+    
+    NSDate* timestamp = location.timestamp;
+    if (fabs([now timeIntervalSinceDate:timestamp]) < 60){
+    }
+    else{
+        NSLog(@"Time 60 more than print");
+        [self makeJSON_Make ];
+        [self stopUpdatingLocationWithMessage:NSLocalizedString(@"Error", @"Error")];
+    }
+}
+
+- (void)stopUpdatingLocationWithMessage:(NSString *)state {
+    
+    [self.locationManager stopUpdatingLocation];
+    self.locationManager.delegate = nil;
+}
+
+-(void)regionStart:(NSString*)str_array{
```

Comment:
this method needs documentation. add an example what `str_array` should be, and how it is transformed. 

e.g.

```
str_array is the region specifier as follows: "region-id:lat:lon:radius"
where 
   region-id  - the region short name, reported back on region enter/exit
   lat - latitude as a float value 
   lon - longitude as a float value
   radius - radius as a float value in (meters?)
```